redwood-lazylayout-compose/src/commonMain/kotlin/app/cash/redwood/lazylayout/compose/LazyListState.kt

```diff
@@ -17,33 +17,27 @@ package app.cash.redwood.lazylayout.compose
 
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableIntStateOf
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.saveable.Saver
 import androidx.compose.runtime.saveable.rememberSaveable
 import androidx.compose.runtime.setValue
 import app.cash.redwood.lazylayout.api.ScrollItemIndex
 
-private const val DEFAULT_PRELOAD_ITEM_COUNT = 15
-private const val SCROLL_IN_PROGRESS_PRELOAD_ITEM_COUNT = 5
-private const val PRIMARY_PRELOAD_ITEM_COUNT = 20
-private const val SECONDARY_PRELOAD_ITEM_COUNT = 10
-
-private const val DEFAULT_SCROLL_INDEX = -1
-
 /**
  * Creates a [LazyListState] that is remembered across compositions.
  */
 @Composable
-public fun rememberLazyListState(): LazyListState {
+public fun rememberLazyListState(
+  strategy: LoadingStrategy = ScrollOptimizedLoadingStrategy(),
+): LazyListState {
   return rememberSaveable(saver = saver) {
-    LazyListState()
+    LazyListState(strategy)
```

Comment:
Seems like `strategy` needs to be added as a key to `rememberSaveable`. Otherwise changes in strategies would not reinitialize the state.